apm-agent-core/src/main/java/co/elastic/apm/agent/impl/ElasticApmTracer.java

```diff
@@ -646,4 +743,26 @@ private String getServiceName(@Nullable ClassLoader initiatingClassLoader) {
     public void resetServiceNameOverrides() {
         serviceNameByClassLoader.clear();
     }
+
+    /**
+     * An enumeration used to represent the current tracer state.
+     */
+    public enum TracerState {
+        /**
+         * Agent is in this state either before starting, or after being stopped.
+         */
+        STOPPED,
```

Comment:
When agent is in `stopped` state after being stopped, can we transition to any other state ?

Reply:
Currently no, there is no such option. This is actually something that bothered me when I introduced the state (the fact that there is no single definition to the `STOPPED` state).
I will add an `UNINITIALIZED` state to distinguish and will add proper commenting.